demos/python_demos/whiteboard_inpainting_demo/utils/ie_tools.py

```diff
@@ -0,0 +1,84 @@
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import cv2
+import logging as log
+import numpy as np
+import os
+import sys
+
+from openvino.inference_engine import IENetwork, IECore # pylint: disable=import-error,E0611
+
+
+class IEModel:
+    """Class for inference of models in the Inference Engine format"""
+    def __init__(self, ie, model_path, conf=.6, device='CPU', ext_path='', num_reqs=1):
+        self.confidence = conf
+        self.load_ie_model(ie, model_path, device, None, ext_path, num_reqs)
+
+    def _preprocess(self, img):
+        _, _, h, w = self.get_input_shape().shape
+        img = np.expand_dims(cv2.resize(img, (w, h)).transpose(2, 0, 1), axis=0)
+        return img
+
+    def forward(self, img):
+        """Performs forward pass of the wrapped IE model"""
+        res = self.net.infer(inputs={self.input_key: self._preprocess(img)})
+        return np.copy(res[self.output_key])
+
+    def get_dedections(self, input):
+        pass
+
+    def get_input_shape(self):
+        """Returns an input shape of the wrapped IE model"""
+        return self.inputs_info[self.input_key]
+
+    def load_ie_model(self, ie, model_xml, device, plugin_dir, cpu_extension='', num_reqs=1):
+        """Loads a model in the Inference Engine format"""
+        model_bin = os.path.splitext(model_xml)[0] + ".bin"
+        # Plugin initialization for specified device and load extensions library if specified
+        log.info("Initializing Inference Engine plugin for %s ", device)
+
+        if cpu_extension and 'CPU' in device:
+            ie.add_extension(cpu_extension, 'CPU')
+        # Read IR
+        log.info("Loading network files:\n\t%s\n\t%s", model_xml, model_bin)
+        net = IENetwork(model=model_xml, weights=model_bin)
+
+        if "CPU" in device:
+            supported_layers = ie.query_network(net, "CPU")
+            not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
+            if not_supported_layers:
+                log.error("Following layers are not supported by the plugin for specified device %s:\n %s",
+                          device, ', '.join(not_supported_layers))
+                log.error("Please try to specify cpu extensions library path in sample's command line parameters using -l "
+                          "or --cpu_extension command line argument")
+                sys.exit(1)
+
+        assert len(net.inputs.keys()) == 1 or len(net.inputs.keys()) == 2, \
```

Comment:
This is strange to have this logic here. Preprocessing and forward pass logic is restricted to support only a single input and single output.

Reply:
`MaskRCNN` has 2 inputs and 4 outputs (in one of previous releases the model had 5 outputs), `SemanticSegmentation` has 1 input and 1 output. So, here I just check that input model has supported format.

Reply:
This is clear. But it's reasonable to place this checks in a corresponding wrapper, not in a base class.

Reply:
Fixed.